br/pkg/storage/s3.go

```diff
@@ -479,6 +471,22 @@ func (rs *S3Storage) WalkDir(ctx context.Context, opt *WalkOption, fn func(strin
 			// you can use the value of the last Key in the response as the marker
 			// in the subsequent request to get the next set of object keys."
 			req.Marker = r.Key
+
+			// when walk on specify directory, the result include storage.Prefix,
+			// which can not be reuse in other API(Open/Read) directly.
+			// so we use TrimPrefix to filter Prefix for next Open/Read.
+			path := strings.TrimPrefix(*r.Key, rs.options.Prefix)
+			itemSize := *r.Size
+
+			// filter out s3's  directries items
+			if itemSize <= 0 {
```

Comment:
`WalkDir` may not be used only by Lightning. To be consistent with normal file system, I think we shouldn't skip empty file in WalkDir.